src/common/admin_socket.cc

```diff
@@ -814,19 +815,24 @@ class RaiseHook: public AdminSocketHook {
         return {{m_cct, fresult, signal_to_send, until}};
       }
 
-      const auto poll_interval = ceph::make_timespan(0.1);
-      auto remaining = (until - ceph::coarse_mono_clock::now());
-      do {
-        using std::chrono::duration_cast;
-        using std::chrono::nanoseconds;
-        std::this_thread::sleep_for(duration_cast<nanoseconds>(std::min(remaining, poll_interval)));
-        if (getppid() != victim) {
-          // suicide if my parent has changed
-          // this means that the original parent process has terminated
-          _exit(1);
+      const ceph::signedspan poll_interval = ceph::make_timespan(0.1);
```

Comment:
ditto (auto type deduction).

Reply:
I wonder why this bit got changed 🤔

Reply:
this whole improvement was about preventing an error that the `auto` has gotten me into. I got an (almost) infinite loop due to the remaining span being deduced as unsigned and comparing greater than zero.

So in this change I decided to avoid ambiguities and spell the types explicitly

Reply:
In that case, please mention the detail in the commit message, otherwise it's a _bit_ hard to infer (now I recall you mentioning this in standup).

Reply:
right! it was clear then but it took some time for us to get back here. I'll update